amazons3/amazons3_utils.bal

```diff
@@ -21,158 +20,332 @@ import ballerina/encoding;
+        if (getObjectHeaders.ifMatch != ()) {
+            requestHeaders[IF_MATCH] = <string>getObjectHeaders.ifMatch;
+        }
+        if (getObjectHeaders.ifNoneMatch != ()) {
+            requestHeaders[IF_NONE_MATCH] = <string>getObjectHeaders.ifNoneMatch;
+        }
+        if (getObjectHeaders.range != ()) {
+            requestHeaders[RANGE] = <string>getObjectHeaders.range;
+        }
+    }
+}
+
+function populateOptionalParameters(string? delimiter = (), string? encodingType = (), int? maxKeys = (), 
+                    string? prefix = (), string? startAfter = (), boolean? fetchOwner = (), 
+                    string? continuationToken = (), map<string> queryParamsMap) returns string {
+    string queryParamsStr = "";
+    // Append query parameter(delimiter).
+    var delimiterStr = delimiter;
+    if (delimiterStr is string) {
+        queryParamsStr = string `${queryParamsStr}&delimiter=${delimiterStr}`;
+        queryParamsMap["delimiter"] = delimiterStr;
+    } 
+
+    // Append query parameter(encoding-type).
+    var encodingTypeStr = encodingType;
+    if (encodingTypeStr is string) {
+        queryParamsStr = string `${queryParamsStr}&encoding-type=${encodingTypeStr}`;
+        queryParamsMap["encoding-type"] = encodingTypeStr;
+    } 
+
+    // Append query parameter(max-keys).
+    var maxKeysVal = maxKeys;
+    if (maxKeysVal is int) {
+        queryParamsStr = string `${queryParamsStr}&max-keys=${maxKeysVal}`;
+        //Check string.convert()
+        queryParamsMap["max-keys"] = io:sprintf("%s", maxKeysVal);
+    } 
+
+    // Append query parameter(prefix).
+    var prefixStr = prefix;
+    if (prefixStr is string) {
+        queryParamsStr = string `${queryParamsStr}&prefix=${prefixStr}`;
+        queryParamsMap["prefix"] = prefixStr;
+    }  
+
+    // Append query parameter(startAfter).
+    var startAfterStr = startAfter;
+    if (startAfterStr is string) {
+        queryParamsStr = string `${queryParamsStr}start-after=${startAfterStr}`;
+        queryParamsMap["start-after"] = startAfterStr;
+    }  
+
+    // Append query parameter(fetch-owner).
+    var fetchOwnerBool = fetchOwner;
+    if (fetchOwnerBool is boolean) {
+        queryParamsStr = string `${queryParamsStr}&fetch-owner=${fetchOwnerBool}`;
+        queryParamsMap["fetch-owner"] = io:sprintf("%s", fetchOwnerBool);
+    }  
+
+    // Append query parameter(continuation-token).
+    var continuationTokenStr = continuationToken;
+    if (continuationTokenStr is string) {
+        queryParamsStr = string `${queryParamsStr}&continuation-token=${continuationTokenStr}`;
+        queryParamsMap["continuation-token"] = continuationTokenStr;
+    } 
+    return queryParamsStr;
+}
+
+function handleResponse(http:Response httpResponse) returns boolean|error {
+    int statusCode = httpResponse.statusCode;
+    if (statusCode == 200 || statusCode == 204) {
+        return true;
+    } else {
+        var amazonResponse = httpResponse.getXmlPayload();
```

Comment:
We can move this block out.

Reply:
This function is used to generate the response for create and delete functions. In that case we won't get any payload in the response for successful creation or deletion. This is used to extract the error message in the error case.

Reply:
What I meant was 
```ballerina
    if (statusCode == 200 || statusCode == 204) {
        return true;
    } 
    var amazonResponse = httpResponse.getXmlPayload();
    if (amazonResponse is xml) {
        return setResponseError(amazonResponse["Message"].getTextValue());
    } else {
        return setResponseError(XML_EXTRACTION_ERROR_MSG);
    }
```